extensions/mqtt/processors/AbstractMQTTProcessor.h

```diff
@@ -256,13 +258,16 @@ class AbstractMQTTProcessor : public core::ProcessorImpl {
   std::optional<std::chrono::seconds> maximum_session_expiry_interval_;
   std::optional<std::chrono::seconds> server_keep_alive_;
 
+  std::shared_ptr<core::RecordSetReader> record_set_reader_;
+  std::shared_ptr<core::RecordSetWriter> record_set_writer_;
```

Comment:
Probably nitpicking but since these only make sense if both or neither are set, maybe we could refactor this into a struct with 2 not null shared ptrs and have an optional member here that could contain that struct.

And since this might be not an  unusual request this struct might even be put into libminifi